### PR TITLE
[Operator Versioning][Edge] Have operator version number available at the loading stage

### PR DESCRIPTION
**Pull request by cccclai**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #67731
* #67728
* #67730
* __->__ #67729

1. operator version is needed to decide whether applying upgrader or not. This pr make it available at loading stage.
2. Swap the order of parsing instruction and operator, because instruction needs to know the operator first because deciding whether applying upgrader or not (change `OP` to `CALL` or not).

Differential Revision: [D32092516](https://our.internmc.facebook.com/intern/diff/D32092516/)

Differential Revision: [D32092516](https://our.internmc.facebook.com/intern/diff/D32092516)

### CODE REVIEW

File: torch/csrc/jit/mobile/import.cpp [hunk 1]
```diff
@@ -375,18 +380,26 @@ void BytecodeDeserializer::parseMethods(
...
-
+    // 1. First pass all operators from models
     parseOperators(
         std::move(ops_list),
         model_version,
         module_load_options_,
         function.get());
 
+    // 2. Decides if upgrader is needed
+    bool use_upgrader =
```
Comment (iseeyuan) [inline torch/csrc/jit/mobile/import.cpp:391]: Is it used?

Comment (iseeyuan) [reply torch/csrc/jit/mobile/import.cpp:391]: It's used in #67731. Would be great to move it there for self-contain.

Comment (cccclai) [reply torch/csrc/jit/mobile/import.cpp:387]: `bool use_upgrader` is a flag and it's decided by the loaded model operator version.. This variable is initialized but not used. One most important change in this pr is to swapping the sequence. Previously `parseInstructions` -> `parseOperators`, now `parseOperators` -> `parseInstructions`.

File: torch/csrc/jit/mobile/import.cpp [hunk 2]
```diff
@@ -376,18 +381,26 @@ void BytecodeDeserializer::parseMethods(
...
+        (operator_version_ < caffe2::serialize::kProducedFileFormatVersion);
+
+    // 3. If upgrader is needed, change change the OP instrunction to CALL
+    // instruction (In next PR, use_upgrader will be parsed to parseInstruction
+    // function and do the actual change)
+    parseInstructions(
```
[the remark this answers is not in the document]
Comment (cccclai) [reply torch/csrc/jit/mobile/import.cpp:393]: Address in the latest commit. `upgrader_mobile.h` will register as function directly, and if upgrader is needed, the instruction will be swapped with CALL operator and it's according function.

**Review by iseeyuan — APPROVED** [review]
LGTM

**Pull request closed without merging**